### ISSUE TITLE
Firebase deploy does not complete from Travis CI

### ISSUE THREAD

**Issue by ramblinjan** [donejs/deploy#46]
See the build logs [here](https://travis-ci.org/donejs/place-my-order#L2785) and [here](https://travis-ci.org/donejs/bitballs/builds/114396465#L4105).  Looking at the Firebase deployment history, it looks like there have not been any Firebase deployments sent from Travis-CI, but have been deployed manually.

When running the deploy script locally, the console output is as follows:

```
donejs - deploying to 'firebase'

donejs - deploying to 'bitballs-donejs'. Please be patient.
donejs - successfully deployed to 'bitballs-donejs'.
```

In the build logs, the success message is missing. It seems like the Travis shell is exiting before the script completes.

**Comment by justinbmeyer** [comment]
Looking into this, the deploy doesn't actually log if there are errors:

```
deploy(packageJSON, packageName).then(function() {
    process.exit(0);
}, function(e){
    process.exit(0);
});
```

when I put a console.log there, I see an error.  We should do something similar to at least let people know what's wrong.

**Comment by daffl** [comment]
The logging error has been fixed in [v0.4.1](https://github.com/donejs/deploy/tree/v0.4.1) and the issues was that we needed to provide a Firebase CI token which you can get via:

```
node_modules/.bin/firebase login:ci
```

And then running

```
travis-encrypt --add -r <your-username>/<your-repository> "FIREBASE_TOKEN=\"<token>\""
```

I updated the guide accordingly in https://github.com/donejs/donejs/pull/565.
